services/galley/src/Galley/Intra/Push.hs

```diff
@@ -107,11 +109,30 @@ data Push
         _pushAsync :: Bool,
         pushOrigin :: UserId,
         pushRecipients :: List1 Recipient,
-        pushJson :: Object
+        pushJson :: Object,
+        pushRecipientListTooLargeToFanout :: Bool
       }
 
 makeLenses ''Push
 
+newPush1Limited :: Bool -> UserId -> PushEvent -> List1 Recipient -> Push
+newPush1Limited recipientListIncomplete from e rr =
```

Comment:
Now we are back to Boolean blindness... :(
What happened to the enum you had defined?